pkg/apis/operator/v1alpha1/types.go

```diff
@@ -91,6 +95,72 @@ type SettingVerticalPodAutoscaler struct {
 	Enabled *bool `json:"enabled,omitempty"`
 }
 
+// VirtualCluster contains configuration for the virtual cluster.
+type VirtualCluster struct {
+	// ETCD contains configuration for the etcds of the virtual garden cluster.
+	// +optional
+	ETCD *ETCD `json:"etcd,omitempty"`
+	// Maintenance contains information about the time window for maintenance operations.
+	Maintenance Maintenance `json:"maintenance"`
+}
+
+// ETCD contains configuration for the etcds of the virtual garden cluster.
+type ETCD struct {
+	// Main contains configuration for the main etcd.
+	// +optional
+	Main *ETCDMain `json:"main,omitempty"`
+	// Events contains configuration for the events etcd.
+	// +optional
+	Events *ETCDEvents `json:"events,omitempty"`
+}
+
+// ETCDMain contains configuration for the main etcd.
+type ETCDMain struct {
```

Comment:
Can you please also add the property `Replicas` to both `EtcdMain` and `EtcdEvents`? Would be necessary if we are to make the virtual garden also run in HA mode, right?

Reply:
Yeah, let's do this in a separate PR. I'm not yet even sure whether this should be configurable in the `Garden` resource, or whether this should be the default behaviour (for the virtual garden, you probably always want HA etcd).

Reply:
I feel there should be a way to control this behavior via the `Garden` resource. Maybe a HA setting like `spec.runtimeCluster.highAvailability: true` and `spec.virtualCluster.highAvailability: true` that can configure HA settings (replicas, scheduling policies, etc) for all the components of runtime and virtual garden cluster that are managed by gardener-operator. WDYT?

Reply:
Yeah, maybe 👍🏻 At least this would be similar to the `Shoot` which is what we are trying to follow as much as possible for `Garden`s as well.